### PR TITLE
feat: new ldk backup server restore

### PR DESCRIPTION
**Pull request by Jasonvdb**
### Description

- Enables backing up and restoring from new backup server.
- Reverts back to deprecated backup server if no backup found on new server

### Linked Issues/Tasks

Does not resolve existing restore issues it should prevent further ones from occurring.
https://github.com/synonymdev/bitkit/issues/713
https://github.com/synonymdev/bitkit/issues/770

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (improving code without creating new functionality)

### Tests

- [ ] Detox test
- [ ] Unit test
- [x] No test

### QA Notes

Backup and restoring lightning funds should be reliable now

### CODE REVIEW

**Comment by limpbrains** [comment]
should performRemoteLdkBackup be removed?

Also list in  Settings -> Backup and Restore screen doesn't represent actual state now. We need a way to get backup info from ldk to update it

File: src/store/actions/backup.ts [hunk 1]
```diff
@@ -179,12 +184,86 @@ export const performRemoteBackup = async <T>({
...
+
+	if (backupSetupRes.isErr()) {
+		return err(backupSetupRes.error);
+	}
+
+	//Check if backup exists on new server
+	const fileListRes = await ldk.backupListFiles();
+	if (fileListRes.isErr()) {
+		return err(fileListRes.error);
+	}
```
Comment (limpbrains) [inline src/store/actions/backup.ts:236]: this throws "Error: Request failed. Missing response from backup server."
I believe we should return { backupExists: false }  in this case

**Review by limpbrains — CHANGES_REQUESTED** [review]

File: src/store/actions/backup.ts (hunk 1 above)
Comment (Jasonvdb) [reply src/store/actions/backup.ts:236]: Might be a server issue here, if there is no backup then the list is just empty. I don't want to assume an error means there's no backup. I'll get back to you on this.

**Comment by Jasonvdb** [comment]
> should performRemoteLdkBackup be removed?

I was thinking let me remove it right after we've done a few rounds of live testing. Just in case.

> Also list in Settings -> Backup and Restore screen doesn't represent actual state now. We need a way to get backup info from ldk to update it

True. Now we don't really have a last backup time to display. I do have a self check function for the backups maybe we can  run this when this view is loaded to show the state.

File: src/store/actions/backup.ts (hunk 1 above)
Comment (limpbrains) [reply src/store/actions/backup.ts:236]: I'm testing with server from development.template. I've created a wallet using master branch, then I'm trying to restore in using new-ldk-backups and I'm getting this error

**Review by limpbrains — APPROVED** [review]

File: src/store/actions/backup.ts (hunk 1 above)
Comment (Jasonvdb) [reply src/store/actions/backup.ts:236]: I've updated the env with new backup server. SSL was required as well as a backend update.

**Pull request merged**